crates/uv-python/src/managed.rs

```diff
@@ -669,6 +655,239 @@ impl ManagedPythonInstallation {
     }
 }
 
+/// A representation of a minor version symlink directory (or junction on Windows)
+/// linking to the home directory of a Python installation.
+#[derive(Clone, Debug)]
+pub struct PythonMinorVersionLink {
```

Comment:
In a sense, the core of the implementation. We use this to derive a symlink directory name from a full Python installation path, to create such a directory (or junction on Windows), and to check for the existence of such a directory. This is also where the preview gating happens (if preview is disabled, the constructor will return `None`).